### ISSUE TITLE
How to run this as a daemon on linux

### ISSUE THREAD

**Issue by ayappant** [dasMulli/dotnet-win32-service#66]
How should I get this working as daemon on linux? Any samples would help

"Build truly portable applications that can for example run as service on windows and as daemon on linux, just using runtime checks / switches"

**Comment by Trapov** [comment]
@ayappant, hi there, I believe you don't need this library if your application only runs on Linux. Here's a simple guide how to do so [(Go to the link)](https://developers.redhat.com/blog/2017/06/07/writing-a-linux-daemon-in-c/)

That means you have to do checks for each platform yourself, at a runtime.
> using runtime checks / switches

And if you really want to target both platforms, then you would write something like this, i guess:

```csharp
if (System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
{
    // your code to handle a control to this library
}
// since Linux doesn't need any daemon-specific code just do your normal code
```

**Comment by dasMulli** [comment]
@ayappant exactly like @Trapov said, a cross-platform "service" would only use this specific library for windows-specific parts.

Running as a linux daemon that starts and stops should already be possible with classic console applications, though more sophisticated control logic is probably blocked on proper signal handling. Also see https://github.com/dotnet/corefx/issues/25384

Closing as out of scope.
